security-analysis/security-analysis-api/src/main/java/com/powsybl/security/condition/AbstractCondition.java

```diff
@@ -0,0 +1,14 @@
+/**
+ * Copyright (c) 2022, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.security.condition;
+
+/**
+ * @author Etienne Lesot <etienne.lesot@rte-france.com>
+ */
+public abstract class AbstractCondition {
+    abstract ConditionType getType();
```

Comment:
If we want the API to be extensible, we need to have a String instead.

By extensible, I mean, allowing a third party to define its own condition type, and handle it in its own simulator.

Do we foresee that kind of usage ?

@geofjamg  @Hadrien-Godard

Reply:
It would be useful if you want, for example, OLF or another tool, to support too specific conditions/actions, that we will not want in the common API.

But OK for me if we don't think it will be needed. If really needed at some point, it would naturally arise and then we should be able to handle that evolution?

In any case, enum or string, all simulators cannot be expected to support all types of conditions and actions of the common API: they will check for unsupported ones in the input.

Reply:
@Hadrien-Godard : do you foresee this for other tools than OLF ?
For OLF, it should be easier to ensure powsybl-core has all the necessary types.

Reply:
No, only for OLF.

Reply:
Ok, so just to summarize for Etienne:

**we keep the enum, here and also for actions**